### PR TITLE
Fix false-positive vision estimation edgecase

### PR DESCRIPTION
**Pull request by amquake**
Reported by @prensing
> there is a problem if the found tags are a valid one plus an invalid tag, so eg it sees a tag but finds noise on the wall, and so lists tags 2 and 15. Lines 90-99 will ignore tag 15, and put 4 corners into the list. It will get past the test on line  100 (there are 4 corners). BUT then at line 106, there are 2 entries in visTags, so it will go to the "else" and try to use solvePNP_SQPNP().

### CODE REVIEW

File: photon-lib/src/main/java/org/photonvision/estimation/VisionEstimation.java [hunk 1]
```diff
@@ -103,7 +103,7 @@ public static PNPResults estimateCamPosePNP(
         Point[] points = OpenCVHelp.cornersToPoints(corners);
 
         // single-tag pnp
-        if (visTags.size() == 1) {
+        if (knownTags.size() == 1) {
```
Comment (mcm001) [inline photon-lib/src/main/java/org/photonvision/estimation/VisionEstimation.java:106]: what happens if knownTags.size = 0?

Comment (HarryXChen3) [reply photon-lib/src/main/java/org/photonvision/estimation/VisionEstimation.java:106]:
```java
if (knownTags.size() == 0 || corners.size() == 0 || corners.size() % 4 != 0) {
    return new PNPResults();
}
```
not involved in this but I assume that this (from earlier) addresses that already

Comment (mcm001) [reply photon-lib/src/main/java/org/photonvision/estimation/VisionEstimation.java:106]: Ah, yep. That's my fault for not actually RTFM.

**Review by mcm001 — APPROVED** [review]

**Comment by mcm001** [comment]
We ball

**Pull request merged**